examples/example-wallet/example_wallet.go

```diff
@@ -0,0 +1,599 @@
+	ExampleWalletArgsTypeKeyDeriver ExampleWalletArgsType = "keyDeriver"
+	ExampleWalletArgsTypeAnyone     ExampleWalletArgsType = "anyone"
+)
+
+type ExampleWalletArgs struct {
+	Type       ExampleWalletArgsType
+	PrivateKey *ec.PrivateKey
+	KeyDeriver *wallet.KeyDeriver
+}
+
+// NewExampleWallet creates a new ExampleWallet from a private key or KeyDeriver
+func NewExampleWallet(rootKeyOrKeyDeriver ExampleWalletArgs) (*ExampleWallet, error) {
+	switch rootKeyOrKeyDeriver.Type {
+	case ExampleWalletArgsTypeKeyDeriver:
+		return &ExampleWallet{
+			keyDeriver: NewKeyDeriver(rootKeyOrKeyDeriver.PrivateKey),
+		}, nil
+	case ExampleWalletArgsTypePrivateKey:
+		return &ExampleWallet{
+			keyDeriver: NewKeyDeriver(rootKeyOrKeyDeriver.PrivateKey),
+		}, nil
+	case ExampleWalletArgsTypeAnyone:
+		// Create an "anyone" key deriver as default
+		kd := NewKeyDeriver(nil)
+		return &ExampleWallet{
+			keyDeriver: kd,
+		}, nil
+	}
+	return nil, errors.New("invalid rootKeyOrKeyDeriver")
+}
+
+func (p *ExampleWallet) GetPublicKey(ctx context.Context, args wallet.GetPublicKeyArgs, _originator string) (*wallet.GetPublicKeyResult, error) {
+	if args.IdentityKey {
+		if p.keyDeriver == nil {
+			return nil, errors.New("keyDeriver is undefined")
+		}
+		return &wallet.GetPublicKeyResult{
+			PublicKey: p.keyDeriver.rootKey.PubKey(),
+		}, nil
+	} else {
+		if args.ProtocolID.Protocol == "" || args.KeyID == "" {
+			return nil, errors.New("protocolID and keyID are required if identityKey is false")
+		}
+
+		if p.keyDeriver == nil {
+			return nil, errors.New("keyDeriver is undefined")
+		}
+
+		// Handle default counterparty (self)
+		counterparty := args.Counterparty
+		if counterparty.Type == wallet.CounterpartyUninitialized {
+			counterparty = wallet.Counterparty{
+				Type: wallet.CounterpartyTypeSelf,
+			}
+		}
+
+		pubKey, err := p.keyDeriver.DerivePublicKey(
+			args.ProtocolID,
+			args.KeyID,
+			counterparty,
+			args.ForSelf,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to derive public key: %v", err)
+		}
+		return &wallet.GetPublicKeyResult{
+			PublicKey: pubKey,
+		}, nil
+	}
+}
+
+// Encrypt encrypts data using the provided protocol ID and key ID
+func (p *ExampleWallet) Encrypt(
+	ctx context.Context,
+	args wallet.EncryptArgs,
+	originator string,
+) (*wallet.EncryptResult, error) {
+
+	if args.Counterparty.Type == wallet.CounterpartyUninitialized {
+		args.Counterparty = wallet.Counterparty{
+			Type: wallet.CounterpartyTypeSelf,
+		}
+	}
+
+	if p.keyDeriver == nil {
+		return nil, errors.New("keyDeriver is undefined")
+	}
+
+	// Create protocol struct from the protocol ID array
+	protocol := args.ProtocolID
+
+	// Handle counterparty
+	counterpartyObj := args.Counterparty
+
+	// Derive a symmetric key for encryption
+	key, err := p.keyDeriver.DeriveSymmetricKey(protocol, args.KeyID, counterpartyObj)
+	if err != nil {
+		return nil, fmt.Errorf("failed to derive symmetric key: %v", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to derive symmetric key: %w", err)
```

Please check other places where you pass error as `%v` instead of `%w`.

Reply:
I followed another approach after @dorzepowski comment - I extended go-sdk proto wallet with some mocks/stubs for missing methods.
So this comment is no longer needed to be fixed.